subprojects/libglnx.wrap

```diff
@@ -0,0 +1,3 @@
+[wrap-git]
+url = https://gitlab.gnome.org/GNOME/libglnx.git
```

Comment:
I think libglnx needs to be a git submodule like it is in the Autotools build system, or a [git subtree](https://github.com/git/git/blob/master/contrib/subtree/git-subtree.txt) like it is in [steam-runtime-tools](https://gitlab.steamos.cloud/steamrt/steam-runtime-tools/). This can be achieved by moving the pre-existing submodule into `subprojects/` and adjusting the Autotools build to cope with that change, like I did in <https://github.com/flatpak/flatpak/pull/4141>. That change can be made as a separate PR that this one would depend on.

Three reasons for this:

* `libglnx` is not a shared library and does not have API or ABI backwards-compatibility guarantees, so we should "pin" it to a known-good version for this version of flatpak-builder instead of always using `HEAD`.
* If we're doing a bug-fix-only release of flatpak-builder (for a serious bug or a security vulnerability) and we want LTS distributions to update their flatpak-builder to the new release, then we won't want that release to include unnecessary updates to `libglnx HEAD` that make the change harder to review.
* Meson automatically includes a submodule or subtree when doing a `meson dist`, even without `--include-subprojects`, preserving the Autotools behaviour that a known-good `libglnx` is bundled in our source releases. We need this because `libglnx` is not designed to be installed system-wide. `meson dist --include-subprojects` would add a copy of `libglnx`, but that option is undesirable because if we start including `subprojects/*.wrap` for projects that *are* designed to normally be installed system-wide, like Flatpak or GLib, we will *not* want to bundle convenience copies of *those* in source releases.

All of these considerations apply more strongly to Flatpak itself than to flatpak-builder, but I think it makes sense for a Meson port of flatpak-builder to follow the same style that we'll want to use in Flatpak.

Reply:
> This can be achieved by moving the pre-existing submodule into subprojects/ and adjusting the Autotools build to cope with that change, like I did in flatpak/flatpak#4141. That change can be made as a separate PR that this one would depend on

#386

Reply:
> If we're doing a bug-fix-only release of flatpak-builder (for a serious bug or a security vulnerability) and we want LTS distributions to update their flatpak-builder to the new release, then we won't want that release to include unnecessary updates to libglnx HEAD that make the change harder to review.
> Meson automatically includes a submodule or subtree when doing a meson dist, even without --include-subprojects, preserving the Autotools behaviour that a known-good libglnx is bundled in our source releases. We need this because libglnx is not designed to be installed system-wide. meson dist --include-subprojects would add a copy of libglnx, but that option is undesirable because if we start including subprojects/*.wrap for projects that are designed to normally be installed system-wide, like Flatpak or GLib, we will not want to bundle convenience copies of those in source releases.

Oh I didn't though about that aspect. I used a wrap file because it was more convenient as meson would fetch the submodule automatically when building for the first time, but now I see the point of having it as submodule then.

Reply:
You can do both. Have a .wrap file which points to the same url as the git submodule. When meson tries to configure the subproject, it will check to see if you're building from a git clone and the subproject is a valid submodule -- if so, it will run `git submodule update --init subprojects/foobar` for you.

This then also works if you're not building from git, but from a tarball that does NOT have the subproject properly disted. meson will simply treat it like a normal wrap-git, and clone it for you.